Sourcery/CodeGenerated/Coding.generated.swift

```diff
@@ -0,0 +1,397 @@
+// Generated using Sourcery 0.5.1 — https://github.com/krzysztofzablocki/Sourcery
+// DO NOT EDIT
+
+import Foundation
+
+extension NSCoder {
+
+    @nonobjc func decode(forKey: String) -> String? {
+        guard let object = self.decodeObject(forKey: forKey) else {
+            return nil
+        }
+
+        return object as? String
+    }
+
+    @nonobjc func decode(forKey: String) -> TypeName? {
+        guard let object = self.decodeObject(forKey: forKey) else {
+            return nil
+        }
+
+        return object as? TypeName
+    }
+
+    @nonobjc func decode(forKey: String) -> AccessLevel? {
+        guard let object = self.decodeObject(forKey: forKey) else {
+            return nil
+        }
+        return object as? AccessLevel
+    }
+
+    @nonobjc func decode(forKey: String) -> Bool {
+        return self.decodeBool(forKey: forKey)
+    }
+
+    @nonobjc func decode(forKey: String) -> Int {
+        return self.decodeInteger(forKey: forKey)
+    }
+
+    func decode<E>(forKey: String) -> E? {
+        guard let object = self.decodeObject(forKey: forKey) else {
+            return nil
+        }
+
+        return object as? E
+    }
+
+}
+
+/*
```

Comment:
Why all that commented out? It does not work? Do we miss something to be able to generate that code properly?